src/component/elements/formik/FormikOnChange.tsx

```diff
@@ -8,11 +8,14 @@ interface FormikOnChangeProps {
 
 const FormikOnChange = (props: FormikOnChangeProps) => {
   const { onChange = () => null, enableValidation = true } = props;
+  const oldValuesRef = useRef({});
   const { values, errors, submitForm, setTouched } = useFormikContext();
-
   useEffect(() => {
     if (enableValidation) {
-      if (Object.keys(errors).length === 0) {
+      if (
+        Object.keys(errors).length === 0 &&
+        JSON.stringify(oldValuesRef.current) !== JSON.stringify(values)
```

Comment:
Why not using directly an empty object rather than a complex useRef ?

Reply:
Because i need to see the previous value and compare the new with the old one

Reply:
the aim of this component is to listen to the formik changes and trigger onChange callback by this i don not need to listen for each input

Reply:
And where to you change the value of oldValuesRef ? There is only 2 instances of this variable in the whole project

![image](https://user-images.githubusercontent.com/1484241/184129841-fda2c248-1728-4a03-8391-c7a810e6f54e.png)